File: 백준/Silver/2217. 로프/로프.swift
import Foundation

let n = Int(readLine()!)!
var weights:[Int] = []

for _ in 0..<n { weights.append(Int(readLine()!)!) }
weights.sort(by: >)

var poss_weights:[Int] = []
for i in 1...n { poss_weights.append(weights[i-1]*i) }
print(poss_weights.max()!)

File: 백준/Silver/2217. 로프/로프.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        
        int N = Integer.parseInt(br.readLine());
        int[] w = new int[N];
        for(int i=0; i<N; i++) w[i] = Integer.parseInt(br.readLine());
        Arrays.sort(w);

        int answer = 0;
        for(int i=1; i<=N; i++) answer = Math.max(answer, w[N-i]*i);
        System.out.println(answer);
    }
}